quinn-proto/src/connection/mod.rs

```diff
@@ -3091,12 +3091,30 @@ impl Connection {
     fn pto_max_path(&self, space: SpaceId) -> Duration {
         match space {
             SpaceId::Initial | SpaceId::Handshake => self.pto(space, PathId::ZERO),
-            SpaceId::Data => self
-                .paths
-                .keys()
-                .map(|path_id| self.pto(space, *path_id))
-                .max()
-                .expect("there should be one at least path"),
+            SpaceId::Data => {
+                // Include paths that have actual RTT measurements:
+                // - validated: PATH_RESPONSE confirmed bidirectional connectivity
+                // - has_samples(): ACK-based RTT sample recorded
+                //
+                // Exclude paths without RTT measurements (unreachable holepunch candidates)
+                // as they inflate the close timer with their default 333ms RTT.
+                let active_pto = self
+                    .paths
+                    .iter()
+                    .filter(|(_, path_state)| {
+                        path_state.data.validated || path_state.data.rtt.has_samples()
```

Comment:
The only working option is the original solution in 233 with only the has_samples check. However it's not a good solution...